realm/realm-library/src/main/java/io/realm/RealmFieldType.java

```diff
@@ -37,7 +37,8 @@
     BINARY(4),
     UNSUPPORTED_TABLE(5),
     UNSUPPORTED_MIXED(6),
-    DATE(7),
+    UNSUPPORTED_DATE(7),
```

Comment:
But having them in user facing code serves no purpose. I can see some merit in mirroring the values, but we can just as well do that in Javadoc instead of actual code?